src/main/java/spoon/reflect/visitor/filter/NamedElementFilter.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.visitor.filter;
+
+import spoon.reflect.declaration.CtNamedElement;
+import spoon.reflect.visitor.Filter;
+
+/**
+ * Filters elements by name and by type (for instance to find a method). Example:
+ *
+ * <pre>
+ * CtMethod&lt;?&gt; normalFor = type.getElements(
```

Comment:
The documentation is wrong: be removing the class type parameter, we also force the type to CtNamedElement, and so we force the user to cast the returned type, even if it's checked. Is that the behaviour we want? Other solution is to remove the simple constructor, to force to specify the expected type (which was specified almost all the time to have the right return type) and then we can keep the class type parameter and the old behaviour.